### DISCUSSION TITLE
Cypress stop working after redirect

### DISCUSSION THREAD

**Discussion by crismarycastellanos** [cypress-io/cypress#28485] [Questions and Help]
After completing a third-party checkout and being redirected, the Cypress sidebar console suddenly stops working. I can't finish the execution of the test because the cypress browser restarted.

I'm using "cypress": "^13.3.0"

https://github.com/cypress-io/cypress/assets/141271700/c143228a-876f-4a8a-84bb-b770c6e09246

This is the same issue https://github.com/cypress-io/cypress/issues/23751#issuecomment-1847509014 Can someone give me a hand? Thanks in advance

**Comment by portyu9** [comment]
Hi @crismarycastellanos 

This appears to be the same Cypress runner/redirect bug tracked in **#23751**, rather than simply a missing `cy.origin()` call. Cypress maintainers were able to reproduce that issue, and it is still open, so there is not a confirmed universal workaround I would recommend.

Since this report uses Cypress `13.3.0`, the first thing I would do today is reproduce it on a current Cypress version. There have been substantial browser, networking, cross-origin, and memory-management changes since Cypress 13.

`cy.origin()` is still required when your test needs to execute Cypress commands after navigating to a different origin, but it will not fix a browser/runner restart caused by the redirect itself.

If it still reproduces on current Cypress:

1. Test with an installed Chrome or Edge browser.
2. Record the exact redirect chain and identify which navigation makes the Cypress runner disappear.
3. Capture browser console/network errors and Cypress debug logs.
4. Add that current reproduction to Cypress issue #23751.

I would also avoid relying on old suggestions such as `experimentalModifyObstructiveThirdPartyCode` as a general fix. That option was tried in the linked bug and did not resolve the problem for affected users.

If the third-party checkout UI itself is outside the behavior your application owns, another practical design is to test your application's handoff to the payment provider and its return/callback separately rather than requiring the entire third-party payment journey in every E2E test.

Please mark this answer as accepted if it helped, thank you.
